### PR TITLE
Add traces to specialize_autograd and lower_grad_of (2nd try)

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

**Review by suo — APPROVED** [review]
accepting to unblock, but looks like there are test failures

File: torch/csrc/jit/ir.cpp [hunk 1]
```diff
@@ -178,8 +178,12 @@ void Node::printAttrValue(std::ostream& out, const Symbol& name) const {
   }
 }
 
-void Node::printAttributes(std::ostream& out, bool ignore_subgraph = false)
-    const {
+void Node::printAttributes(std::ostream &out, bool ignore_subgraph = false,
+                           bool skip) const {
+
+  if (skip) {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/ir.cpp]: This looks strange, can we just check if we want to call `printAttributes` at the callsites?

**Review by ZolotukhinM — APPROVED** [review]
Looks good to me!

File: torch/csrc/jit/ir.h [hunk 2]
```diff
@@ -734,9 +734,9 @@ struct TORCH_API Node {
     return getAttr<TensorsAttr>(name);
   }
 
- private:
+private:
   void printAttrValue(std::ostream& out, const Symbol& name) const;
-  void printAttributes(std::ostream& out, bool ignore_subgraph) const;
+  void printAttributes(std::ostream &out, bool ignore_subgraph) const;
```
Comment (ZolotukhinM) [inline torch/csrc/jit/ir.h]: Clang-format? :)

File: torch/csrc/jit/passes/specialize_autogradzero.cpp [hunk 3]
```diff
@@ -97,7 +111,10 @@ void specializeAutogradZero(Graph& g) {
           auto* add_output = add_node->output();
           state[add_output] = State::Nonzero;
           n->output()->replaceAllUsesWith(add_output);
+          GRAPH_UPDATE("Simplifying ", getHeader(n), " to ",
+                       getHeader(add_node));
           it.destroyCurrent();
+
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/specialize_autogradzero.cpp]: Nit: unnecessary change.
